rust_src/remacs-sys/lib.rs

```diff
@@ -924,10 +924,25 @@ pub struct Lisp_Frame {
     // frame.foo the proper method is fget_foo(frame).
 }
 
+#[repr(C)]
+pub enum OutputMethod {
+    output_initial,
+    output_termcap,
+    output_x_window,
+    output_w32,
+    output_ns,
+}
+
+#[repr(C)]
+pub struct terminal {}
+
 /// Functions to access members of `struct frame`.
+#[allow(improper_ctypes)]
 extern "C" {
     pub fn fget_column_width(f: *const Lisp_Frame) -> c_int;
     pub fn fget_line_height(f: *const Lisp_Frame) -> c_int;
+    pub fn fget_terminal(f: *const Lisp_Frame) -> *const terminal;
+    pub fn fget_output_method(f: *const Lisp_Frame) -> OutputMethod;
```

Comment:
This is, unfortunately, not sound. See the announcement of bindgen 0.31, which changed from representing C enums as Rust enums: https://www.reddit.com/r/rust/comments/795n0k/announcing_bindgen_0310/

> C/C++ enums are now translated into constants by default, rather than Rust enums. The old behavior was a big footgun because rustc assumes that the only values of an enum are its variants, whereas a lot of C/C++ code uses random values as enums. Put these two things and it leads to undefined behavior.

Changing this will also fix the need to include an "unnecessary" match pattern below.